java/ql/src/Telemetry/SupportedExternalSources.ql

```diff
@@ -0,0 +1,22 @@
+/**
+ * @name Supported sources in external libraries
+ * @description A list of 3rd party APIs detected as sources. Excludes test and generated code.
+ * @id java/telemetry/supported-external-api-sources
+ * @kind metric
+ * @metricType callable
+ */
+
+import java
+import APIUsage
+import ExternalAPI
+import semmle.code.java.GeneratedFiles
+
+from ExternalAPI api
+where
+  not api.isTestLibrary() and
+  supportKind(api) = "source"
+select api.asCSV(api) as csv,
+  strictcount(Call c |
+    c.getCallee() = api and
+    not c.getFile() instanceof GeneratedFile
+  ) as Usages order by Usages desc
```

Comment:
Minor style nit: introduce `int usages` in the `from` clause and define it as the count in the `where` clause.